Assets/HoloToolkit/Utilities/Editor/SceneLauncherEditor.cs

```diff
@@ -0,0 +1,142 @@
+        private SerializedProperty buttonRowMaxProperty;
+
+        private void OnEnable()
+        {
+            sceneMappingProperty = serializedObject.FindProperty("sceneMapping");
+            buttonSpawnLocationProperty = serializedObject.FindProperty("ButtonSpawnLocation");
+            buttonPrefabProperty = serializedObject.FindProperty("SceneButtonPrefab");
+            buttonRowMaxProperty = serializedObject.FindProperty("MaxRows");
+
+            CheckBuildScenes(sceneMappingProperty);
+        }
+
+        public override void OnInspectorGUI()
+        {
+            serializedObject.Update();
+            EditorGUILayout.PropertyField(buttonSpawnLocationProperty);
+            EditorGUILayout.PropertyField(buttonPrefabProperty);
+            EditorGUILayout.IntSlider(buttonRowMaxProperty, 1, 10);
+            CheckBuildScenes(sceneMappingProperty);
+            ShowSceneList(sceneMappingProperty);
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void ShowSceneList(SerializedProperty sceneList)
+        {
+            // property name with expansion widget
+            EditorGUILayout.PropertyField(sceneList);
+
+            if (EditorBuildSettings.scenes.Length == 0)
+            {
+                sceneList.ClearArray();
+
+                int index = sceneList.arraySize;
+
+                sceneList.InsertArrayElementAtIndex(index);
+                sceneList.GetArrayElementAtIndex(index).FindPropertyRelative("ScenePath").stringValue = SceneManager.GetActiveScene().path;
+                sceneList.GetArrayElementAtIndex(index).FindPropertyRelative("IsButtonEnabled").boolValue = true;
+
+                EditorBuildSettings.scenes = new[] { new EditorBuildSettingsScene(SceneManager.GetActiveScene().path, true) };
+            }
+
+            if (sceneList.isExpanded)
+            {
+                EditorGUI.indentLevel++;
+
+                // Scene rows
+                for (int i = 0; i < sceneList.arraySize; i++)
+                {
+                    EditorGUILayout.BeginHorizontal();
+                    // Disable the toggle if the scene is not enabled in the build settings.
+                    GUI.enabled = EditorBuildSettings.scenes[i].enabled;
+
+                    EditorGUILayout.PropertyField(sceneList.GetArrayElementAtIndex(i).FindPropertyRelative("IsButtonEnabled"));
+
+                    GUI.enabled = false;
+
+                    string path = sceneList.GetArrayElementAtIndex(i).FindPropertyRelative("ScenePath").stringValue;
+                    var sceneAsset = AssetDatabase.LoadAssetAtPath(path, typeof(SceneAsset)) as SceneAsset;
+                    EditorGUILayout.ObjectField(sceneAsset, typeof(SceneAsset), false);
+
+                    GUI.enabled = true;
+                    EditorGUILayout.EndHorizontal();
+                }
+
+                EditorGUI.indentLevel--;
+            }
+
+        }
+
+        private void CheckBuildScenes(SerializedProperty list)
```

Comment:
When I open up the build settings, add a scene, and then Run in Unity then a button for the added scene doesn't show up. First I have to select the SceneLauncher prefab in order to display it in the inspector in order to get the list of buttons to update. Can we try including a [PostProcessScene](https://docs.unity3d.com/ScriptReference/Callbacks.PostProcessSceneAttribute.html) attribute that calls CheckBuildScenes so that we ensure that this update has when the app is built or run?

Reply:
Great idea

Reply:
There wasn't an easy way to use `PostProcessScene` attribute because it has to be a static call to the class, and our serialized properties could still be unset.  We would somehow still need to know about our prefab somehow, even if it's in the scene.

Reply:
Btw, by default if a new scene is added it's automatically disabled unless the user goes and enables it, so I think we would want people to update this prefab anyway.  Updating the build settings, doing the post process, and then building wouldn't have made the buttons show up anyway.